Sources/Rendering/Core/ScalarBarActor/index.js

```diff
@@ -682,12 +685,27 @@ function vtkScalarBarActor(publicAPI, model) {
     mat4.transpose(cmat, cmat);
     mat4.invert(invmat, cmat);
 
-    const haveExtraColors =
-      typeof model.scalarsToColors.getNanColor === 'function' &&
-      typeof model.scalarsToColors.getAboveRangeColor === 'function' &&
-      typeof model.scalarsToColors.getBelowRangeColor === 'function';
+    let numberOfExtraColors = 0;
+    if (
+      model.drawNanAnnotation &&
+      typeof model.scalarsToColors.getNanColor === 'function'
+    ) {
+      numberOfExtraColors += 1;
+    }
+    if (
+      model.drawBelowRangeSwatch &&
+      typeof model.scalarsToColors.getBelowRangeColor === 'function'
+    ) {
+      numberOfExtraColors += 1;
+    }
+    if (
+      model.drawAboveRangeSwatch &&
+      typeof model.scalarsToColors.getAboveRangeColor === 'function'
+    ) {
+      numberOfExtraColors += 1;
+    }
 
-    const numPts = 4 + (haveExtraColors ? 12 : 0);
+    const numPts = 4 * (1 + numberOfExtraColors);
```

Comment:
odd, it used to be `4 * 12` and now it is `4 * 4`

Reply:
It was `4+12` before, now it's `4+4*numberOfExtras`

Reply:
my bad, did not read correctly